Userland/Libraries/LibWeb/Streams/AbstractOperations.cpp

```diff
@@ -247,4 +376,224 @@ WebIDL::ExceptionOr<void> set_up_readable_stream_default_reader(ReadableStreamDe
     return {};
 }
 
+// https://streams.spec.whatwg.org/#readable-stream-default-controller-close
+void readable_stream_default_controller_close(ReadableStreamDefaultController& controller)
+{
+    // 1. If ! ReadableStreamDefaultControllerCanCloseOrEnqueue(controller) is false, return.
+    if (!readable_stream_default_controller_can_close_or_enqueue(controller))
+        return;
+
+    // 2. Let stream be controller.[[stream]].
+    auto stream = controller.stream();
+
+    // 3. Set controller.[[closeRequested]] to true.
+    controller.set_close_requested(true);
+
+    // 4. If controller.[[queue]] is empty,
+    if (controller.queue().is_empty()) {
+        // 1. Perform ! ReadableStreamDefaultControllerClearAlgorithms(controller).
+        readable_stream_default_controller_clear_algorithms(controller);
+
+        // 2. Perform ! ReadableStreamClose(stream).
+        readable_stream_close(*stream);
+    }
+}
+
+// https://streams.spec.whatwg.org/#readable-stream-default-controller-enqueue
+WebIDL::ExceptionOr<void> readable_stream_default_controller_enqueue(ReadableStreamDefaultController& controller, JS::Value chunk)
+{
+    // 1. If ! ReadableStreamDefaultControllerCanCloseOrEnqueue(controller) is false, return.
+    if (!readable_stream_default_controller_can_close_or_enqueue(controller))
+        return {};
+
+    // 2. Let stream be controller.[[stream]].
+    auto stream = controller.stream();
+
+    // 3. If ! IsReadableStreamLocked(stream) is true and ! ReadableStreamGetNumReadRequests(stream) > 0, perform ! ReadableStreamFulfillReadRequest(stream, chunk, false).
+    if (is_readable_stream_locked(*stream) && readable_stream_get_num_read_requests(*stream) > 0) {
+        readable_stream_fulfill_read_request(*stream, chunk, false);
+    }
+    // 4. Otherwise,
+    else {
+        // 1. Let result be the result of performing controller.[[strategySizeAlgorithm]], passing in chunk, and interpreting the result as a completion record.
+        auto result = controller.strategy_size_algorithm().value()(chunk);
```

Comment:
Prefer `*` over `value()`

Reply:
Even though it requires an extra set of parens? imo value() is cleaner but I can still make the change